services/src/main/java/org/keycloak/services/clienttype/client/TypeAwareClientModelDelegate.java

```diff
@@ -216,50 +216,23 @@ public void removeRedirectUri(String redirectUri) {
 
     @Override
     public void setAttribute(String name, String value) {
-        TypedClientExtendedAttribute attribute = TypedClientExtendedAttribute.getAttributesByName().get(name);
-        if (attribute != null) {
-            attribute.setClientAttribute(clientType, value, (newValue) -> super.setAttribute(name, newValue), String.class);
-        } else {
+        if (clientType.isApplicable(name)) {
             super.setAttribute(name, value);
         }
     }
 
     @Override
     public void removeAttribute(String name) {
-        TypedClientExtendedAttribute attribute = TypedClientExtendedAttribute.getAttributesByName().get(name);
-        if (attribute != null) {
-            attribute.setClientAttribute(clientType, null, (val) -> super.removeAttribute(name), String.class);
-        } else {
-            super.removeAttribute(name);
-        }
+        super.removeAttribute(name);
     }
 
     @Override
     public String getAttribute(String name) {
-        TypedClientExtendedAttribute attribute = TypedClientExtendedAttribute.getAttributesByName().get(name);
-        if (attribute != null) {
-            return attribute.getClientAttribute(clientType, String.class);
-        } else {
-            return super.getAttribute(name);
-        }
+        return super.getAttribute(name);
```

Comment:
Doesn't this change means that attribute defined on clientType won't be considered at all (as it will always delegate to client)? AFAIK we rather prefer that value defined on client should be preferred though?

Reply:
+1 to adjust in later PR

Reply:
@vickeybrown FYI. introduced https://github.com/keycloak/keycloak/issues/30214 and https://github.com/keycloak/keycloak/issues/30216 as a follow-ups